packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/log_query_and_deprecation.test.ts

```diff
@@ -616,5 +711,221 @@ describe('instrumentQueryAndDeprecationLogger', () => {
         /Query:\n.*200\n.*GET \/_path\?hello\=dolly/
       );
     });
+
+    describe('Request body redaction on some APIs', () => {
+      it('redacts for an API in the extended list (path only)', () => {
+        instrumentEsQueryAndDeprecationLogger({
+          logger,
+          client,
+          type: 'test type',
+          apisToRedactInLogs: [{ path: '/foo' }],
+        });
+
+        const response = createApiResponse({
+          body: {},
+          statusCode: 200,
+          headers: {},
+          params: {
+            method: 'GET',
+            path: '/foo',
+            querystring: { hello: 'dolly' },
+            body: {
+              seq_no_primary_term: true,
+              query: {
+                term: { user: 'kimchy' },
+              },
+            },
+          },
+        });
+
+        client.diagnostic.emit('response', null, response);
+        expect(loggingSystemMock.collect(logger).debug[0][0]).toMatchInlineSnapshot(`
+          "200
+          GET /foo?hello=dolly
+          [redacted]"
+        `);
+      });
+
+      it('redacts for an API in the extended list (method and path)', () => {
+        instrumentEsQueryAndDeprecationLogger({
+          logger,
+          client,
+          type: 'test type',
+          apisToRedactInLogs: [{ method: 'GET', path: '/foo' }],
+        });
+
+        const response = createApiResponse({
+          body: {},
+          statusCode: 200,
+          headers: {},
+          params: {
+            method: 'GET',
+            path: '/foo',
+            querystring: { hello: 'dolly' },
+            body: {
+              seq_no_primary_term: true,
+              query: {
+                term: { user: 'kimchy' },
+              },
+            },
+          },
+        });
+
+        client.diagnostic.emit('response', null, response);
+        expect(loggingSystemMock.collect(logger).debug[0][0]).toMatchInlineSnapshot(`
+          "200
+          GET /foo?hello=dolly
+          [redacted]"
+        `);
+      });
+
+      it('does not redact for an API in the extended list when method does not match', () => {
```

Comment:
we lack a test for not redacting when the path isn't in the extended list

Reply:
IMO, this test tests both: it's not in the extended list, nor the hardcoded one. Afterall, the extended list _extends_ the built-in one.

I can rephrase the test if you think it's worth it.

Reply:
I mean we need a test like 
> does not redact for an API in the extended list when **path** does not match

Reply:
Oh! Got it! Pushed in [`4dbd384` (#153049)](https://github.com/elastic/kibana/pull/153049/commits/4dbd384b4221940e7756078ef7a67f82b8556889). I also added an extra test 😇